### ISSUE TITLE
Deprecate the old varlink for podman-env

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#9230]
Podman v2 doesn't support the varlink protocol any longer, so have to move to using the unix socket over ssh:

```
export CONTAINER_HOST="ssh://docker@192.168.99.100:22/run/podman/podman.sock"
export CONTAINER_SSHKEY=/home/anders/.minikube/machines/minikube/id_rsa
```

This also means that we need to set up the `podman.socket` unit, in order to start the `podman.service` unit.

As well as making the `/run/podman/podman.sock` available to the connecting user, in this case called "docker".

See https://github.com/containers/podman/issues/6809 for configuration

Requirements: Podman 2.0.6 or later

Currently we use PODMAN_VARLINK_BRIDGE and `sudo varlink -A 'podman varlink \$VARLINK_ADDRESS' bridge`

This is only available with podman 1.9.3, which is no longer supported upstream (we have to build our own podman-remote)

See #8596

We can time this podman-env `ssh:// ` feature with docker-env: #9229

~~Could also take a look at using `podman system connection` perhaps.~~

~~https://github.com/containers/podman/issues/7602#issuecomment-691434579~~

Note: Podman 3.0 will **remove** varlink support

It is usually compiled out, even with the 2.x builds

**Comment by afbjorklund** [comment]
Implemented in PR #9535

**Comment by afbjorklund** [comment]
Was released in 1.15.0
